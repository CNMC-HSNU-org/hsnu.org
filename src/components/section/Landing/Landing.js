import React from "react"
import Jumbotron from "react-bootstrap/Jumbotron"
import { useStaticQuery, graphql } from "gatsby"

// style
import "./Landing.scss"

const Landing = () => {
  const headline = useStaticQuery(graphql`
    {
      allWordpressWpNews(limit: 1, sort: { fields: date, order: DESC }) {
        edges {
          node {
            title
            acf {
              image {
                source_url
              }
              link
            }
          }
        }
      }
    }
  `).allWordpressWpNews.edges[0]

  return (
    <Jumbotron fluid>
      <img
        src={headline.node.acf.image.source_url + ".webp"}
        alt={"Headline"}
      ></img>
      <div className={"fade-layer"}></div>
      <a href={headline.node.acf.link}>
        <h1 className={"is-1 serif bold"}>{headline.node.title}</h1>
      </a>
    </Jumbotron>
  )
}

export default Landing
