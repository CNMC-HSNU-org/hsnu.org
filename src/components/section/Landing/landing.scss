.jumbotron {
  // position
  position: relative;

  // box
  height: 70vh;
  width: 100%;
  padding: 0;
  margin: 0;

  // background (image)
  background-image: url("../../../images/cover/fire.jpg");
  background-size: cover;
  background-position: center;

  // title
  a {
    // position
    display: inline-block;
    position: absolute;
    top: 75%;
    left: 50%;
    transform: translate(-50%, -50%);

    // box
    width: 100%;
    padding: 0 1rem;
    color: white;

    // limit line
    overflow: hidden;
    text-overflow: ellipsis;
    display: -webkit-box;
    -webkit-box-orient: vertical;
    -webkit-line-clamp: 3; /* number of lines to show */
  }
  a:hover {
    cursor: pointer;
    color: white;
  }

  // fade (component)
  .fade-layer {
    position: absolute;
    bottom: 0;
    height: 40%;
    width: 100%;
    background: linear-gradient(
      180deg,
      rgba(0, 0, 0, 0) 0%,
      rgba(0, 0, 0, 0.6) 28.65%,
      #000000 100%
    );
  }
}

@media (min-width: 992px) {
  .jumbotron {
    // box
    height: 100vh;

    // title
    a {
      width: 60%;
    }
  }
}
