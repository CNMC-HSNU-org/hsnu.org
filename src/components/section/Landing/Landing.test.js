import React from "react"
import renderer from "react-test-renderer"
import { LandingPure } from "./Landing"

describe("Section/Landing", () => {
  it("renders correctly", () => {
    const headline = {
      node: {
        title:
          "樂園的選擇：從言論與資訊自由被剝奪，看亡國感的生成 | The reporter",
        acf: {
          image: {
            sizes: {
              wordpress_1536x1536:
                "https://wordpress.hsnu.org/wp-content/uploads/2020/03/20191126170557-6072c73cbfb5c0301e948f36af2cf1e0-desktop.jpg",
            },
          },
          link:
            "https://www.twreporter.org/a/bookreview-counter-attack-on-the-fear-of-demising-nation",
        },
      },
    }
    const tree = renderer.create(<LandingPure headline={headline} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
