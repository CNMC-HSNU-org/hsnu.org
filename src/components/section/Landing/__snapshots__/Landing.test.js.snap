// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section/Landing renders correctly 1`] = `
<div
  className="jumbotron jumbotron-fluid"
  id="landing"
>
  <img
    alt="Headline"
    rel="preload"
    src="https://wordpress.hsnu.org/wp-content/uploads/2020/03/20191126170557-6072c73cbfb5c0301e948f36af2cf1e0-desktop.jpg"
  />
  <div
    className="fade-layer"
  />
  <a
    href="https://www.twreporter.org/a/bookreview-counter-attack-on-the-fear-of-demising-nation"
  >
    <h1
      className="is-1 serif"
    >
      樂園的選擇：從言論與資訊自由被剝奪，看亡國感的生成 | The reporter
    </h1>
  </a>
</div>
`;
